src/node_crypto.cc

```diff
@@ -2773,6 +2775,38 @@ void SSLWrap<Base>::GetSharedSigalgs(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <class Base>
+void SSLWrap<Base>::ExportKeyingMaterial(
+                                      const FunctionCallbackInfo<Value>& args
+                                      ) {
+  CHECK(args[0]->IsInt32());
+  CHECK(args[1]->IsString());
```

Comment:
This method is exposed directly to user JS code so you should be prepared to handle bad inputs. Can you add tests for various bad inputs?

You can write a JS wrapper and do validation there, then it's okay to leave these as CHECKs.

Reply:
thanks for the hint, I added parameter validation and added tests for parameters. I also found out that it crashes if the tls socke tis not secure established, added handling and a test case for this too.